python.d/python_modules/third_party/mcrcon.py

```diff
@@ -0,0 +1,90 @@
+# Minecraft Remote Console module.
+#
+# Copyright (C) 2015 Barnaby Gale
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies of the Software and its documentation and acknowledgment shall be
+# given in the documentation and software packages that this Software was
+# used.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import socket
+import select
+import struct
+import time
+
+
+class MCRconException(Exception):
+    pass
+
+
+class MCRcon(object):
+    socket = None
+
+    def connect(self, host, port, password):
+        if self.socket is not None:
+            raise MCRconException("Already connected")
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.socket.connect((host, port))
+        self.send(3, password)
+
+    def disconnect(self):
+        if self.socket is None:
+            raise MCRconException("Already disconnected")
+        self.socket.close()
+        self.socket = None
+
+    def read(self, length):
+        data = b""
+        while len(data) < length:
+            data += self.socket.recv(length - len(data))
+        return data
+
+    def send(self, out_type, out_data):
+        if self.socket is None:
+            raise MCRconException("Must connect before sending data")
+
+        # Send a request packet
+        out_payload = struct.pack('<ii', 0, out_type) + out_data.encode('utf8') + b'\x00\x00'
+        out_length = struct.pack('<i', len(out_payload))
+        self.socket.send(out_length + out_payload)
+
+        # Read response packets
+        in_data = ""
+        while True:
+            # Read a packet
+            in_length, = struct.unpack('<i', self.read(4))
+            in_payload = self.read(in_length)
+            in_id, in_type = struct.unpack('<ii', in_payload[:8])
```

Comment:
unused variable `in_type`

Reply:
Huh, that it is.  I hadn't really looked at this code in depth, just pulled it over from the upstream repo.  I'll get that taken care of.